samplers/aws/xray/internal/manifest.go

```diff
@@ -0,0 +1,359 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	crypto "crypto/rand"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jinzhu/copier"
+
+	"go.opentelemetry.io/contrib/samplers/aws/xray/internal/util"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+const defaultInterval = int64(10)
+
+const manifestTTL = 3600
+
+// Manifest represents a full sampling ruleset and provides
+// option for configuring Logger, Clock and xrayClient.
+type Manifest struct {
+	Rules                          []Rule
+	SamplingTargetsPollingInterval time.Duration
+	refreshedAt                    int64
+	xrayClient                     *xrayClient
+	clientID                       *string
+	logger                         logr.Logger
+	Clock                          util.Clock
+	mu                             sync.RWMutex
+}
+
+// NewManifest return manifest object configured with logging, clock and xrayClient.
+func NewManifest(addr string, logger logr.Logger) (*Manifest, error) {
+	// generate client for getSamplingRules and getSamplingTargets API call
+	client, err := newClient(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	// generate clientID for sampling statistics
+	clientID, err := generateClientID()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Manifest{
+		xrayClient:                     client,
+		Clock:                          &util.DefaultClock{},
+		logger:                         logger,
+		SamplingTargetsPollingInterval: 10 * time.Second,
+		clientID:                       clientID,
+	}, nil
+}
+
+// Expired returns true if the manifest has not been successfully refreshed in
+// manifestTTL seconds.
+func (m *Manifest) Expired() bool {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	return m.refreshedAt < m.Clock.Now().Unix()-manifestTTL
+}
+
+// MatchAgainstManifestRules returns a Rule and boolean flag set as true if rule has been match against span attributes, otherwise nil and false
```

Comment:
```suggestion
// MatchAgainstManifestRules returns a Rule and boolean flag set as true 
// if rule has been match against span attributes, otherwise nil and false.
```